meta-mender-core/recipes-mender/mender-client/files/mender-client-resize-data-part.sh.in

```diff
@@ -42,3 +42,9 @@ fi
 echo "w" | fdisk ${mender_storage_device} &> /dev/null || true
 
 /usr/sbin/parted -s ${mender_storage_device} resizepart @MENDER_DATA_PART_NUMBER@ 100%
+
+# Modifying the partition table in any way may trigger udev at some point.
+# We don't want to continue right while udev is recreating /dev/disk symlinks.
+# Instead, we want this to happen deterministically now and be done when the script is done.
+/usr/sbin/partprobe
+/sbin/udevadm settle
```

Comment:
I wonder if this should perhaps be conditional. According to [the Yocto Manual](https://docs.yoctoproject.org/singleindex.html#using-devtmpfs-and-a-device-manager) you can set the dev manager to something other than udev. Perhaps surround this in a `if [ -x /sbin/udevadm ]`?